pyaedt/filtersolutions_core/__init__.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import sys
+
+from pyaedt.filtersolutions_core.dll_interface import DllInterface
+
+# Store the current module in a variable for easy access and modification within the module itself.
+_this = sys.modules[__name__]
+
+# Initialize the internal DLL interface attribute to None. This will be set to an actual
+# DllInterface instance when needed, implementing a lazy initialization pattern.
+_this._internal_dll_interface = None
+
+
+def _dll_interface(version=None) -> DllInterface:
```

Comment:
@ramin4667 The version is only used the first time this method is called. If someone tries to provide a version again in a subsequent call it won't be used. That is the behavior we want. However, that might be confusing to someone using the method - they might think they can switch versions or something like that. Let's raise an exception when this method is called with a version specified other than the one that's already loaded.